.devcontainer/Containerfile

```diff
@@ -0,0 +1,34 @@
+FROM python:3.13-slim
+
+ARG USERNAME=vscode
+ARG USER_UID=1000
+ARG USER_GID=$USER_UID
+
+RUN apt-get update && export DEBIAN_FRONTEND=noninteractive \
+    && apt-get -y install --no-install-recommends \
+        git \
+        curl \
+        wget \
+        build-essential \
+        sudo \
+        libffi-dev \
+    && groupadd --gid $USER_GID $USERNAME \
+    && useradd --uid $USER_UID --gid $USER_GID -m $USERNAME \
+    && echo $USERNAME ALL=\(root\) NOPASSWD:ALL > /etc/sudoers.d/$USERNAME \
+    && chmod 0440 /etc/sudoers.d/$USERNAME \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+USER $USERNAME
+
+RUN pip install --user pipx \
+    && python -m pipx ensurepath
+
+ENV PATH="/home/${USERNAME}/.local/bin:${PATH}"
+
+RUN pipx install hatch \
+    && pipx install uv
```

Comment:
Minor nit: I would make this a bit more straightforward:

1. Install `uv` (probably easiest through their `curl ... | sh` script)
2. Install `hatch` through `uv` with: `uv tool install hatch`